src_assets/common/assets/web/index.html

```diff
@@ -107,21 +107,27 @@ <h2>Legal</h2>
         let v = this.githubVersion.name;
         // If the version starts with a v, remove it
         if (v.indexOf("v") === 0) v = v.substring(1);
+
+        // if nightly, we do an additional check to make sure its an actual upgrade.
+        if (this.version.split('.').length > 3) {
+          const stableVersion = this.version.split('.').slice(0, 3).join('.');
+          return this.githubVersion.tag_name.substring(1) > stableVersion;
+        }
+
         // return true if the version is different, otherwise false
         return v !== this.version.split(".")[0];
       },
-      // Check for new nightly version
       hasNewNightly() {
         // If we're not on a nightly build, just return false
         // If length of version split is 3, we're on a stable build
         if (!this.version || !this.nightlyData || this.version.split(".").length === 3) return false;
         // If built with dirty git tree, return false
         if (this.version.indexOf("dirty") !== -1) return false;
         // Get the commit hash
-        let commit = this.version.split(".")[-1];
+        let commit = this.version.split(".").pop();
```

Comment:
Using pop changes the original array and ends up going crazy.

![image](https://user-images.githubusercontent.com/42013603/227726850-98c87fab-8222-4fe9-98b2-983673e10f70.png)

This part of the code was never hit before though, due to a mistake when I finalized the PR that added the versioning notifications.

L92 needs to change to this.

`if (this.version.split(".").length === 4) {`

Elix was splitting the version with `-g`, but I had to change it to `.` to satisfy some build systems like Archlinux/AUR... and I must have missed changing that.

Reply:
You’re right, this is more of an error on my part for not double checking that. It is best practices to make computed properties immutable, meaning if you mutated the value it doesn’t change the original.

That’s normally how it’s done on my projects anyway and due to my on conventions I made an incorrect assumption.

The fix is pretty simple either way.

Reply:
Actually looking at that, pop shouldn’t be changing original version here. I thought it was a computed property, oops.

Can you explain your setup so I can see if I can replicate this bug?

Reply:
My screenshot shows what happens when L92 is fixed AND using the `pop`. If using the `[-1]` and fixing L92, the code works correctly.

This screenshot, below, is strictly changing L92 (doesn't include your fix for stable).

![image](https://user-images.githubusercontent.com/42013603/227731374-985acd52-151e-4ab6-9791-7f9d4918b84a.png)

And without fixing L92, it just shows the "new stable version" message.

Reply:
I know, but I’m trying to figure out how you’re seeing that as it doesn’t happen on my side. 

The -1 does not work in JavaScript. That code statement is busted, it will return back undefined in all cases. 

Basically changing it to pop() doesn’t cause it to break, but rather some other lines of code in the existing code has a bug I can’t replicate on my side.

Reply:
Not sure what happened to cause the versioning to get all messed up in my original screenshot. `pop` seems to be okay now.

We still need to address L92 though.

Reply:
I thought I had included that fix already in my commits, not sure why it reverted back. I've pushed those requested changes as well as I refactored the names of the computed properties to reflect their intention as well, which would eliminate the purpose of the comments of the function itself.